File: ocho/src/main/java/ocho/entity/Order.java
package ocho.entity;

import java.math.BigInteger;
import java.util.Date;
import java.util.Map;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document
public class Order {
    public static String PENDING = "Pendiente";
    public static String APROVED = "Aprobada";
    public static String REJECTED = "Rechazada";
    
    @Id
    private BigInteger id;

    private Date registerDay;

    private String status;

    private User salesMan;

    private Map<Integer, Laptop> products;

    private Map<Integer, Integer> quantities;

}

File: ocho/view/src/api/order.ts
import { CRUD } from "./crud";
import type { Record as Base } from "./crud";
import type { HttpSession } from "./session";
import type { Record as User } from "./user";
import type { Record as Laptop } from "./laptop";

export default class OrderCRUD extends CRUD<Record> {
  constructor(session: HttpSession) {
    super(session, "order");
  }
}

/**
 * Types
 */
export interface Record extends Order, Base {}

export interface Order {
  registerDay: string;
  status: string;
  salesMan: User;
  products: Laptop[];
  quantities: {
    [K: string]: number;
  };
}
